sdk/python/tests/compiler/testdata/recursion_while.yaml

```diff
@@ -0,0 +1,129 @@
+      "artifacts/$PIPELINERUN/flip-test-2/output.tgz", "name": "flip-test-2-output",
+      "path": "/tmp/output"}], "flip-test-3": [{"key": "artifacts/$PIPELINERUN/flip-test-3/output.tgz",
+      "name": "flip-test-3-output", "path": "/tmp/output"}]}'
+  name: pipeline-flip-coin
+spec:
+  params:
+  - name: maxVal
+    value: '12'
+  pipelineSpec:
+    params:
+    - default: '12'
+      name: maxVal
+    tasks:
+    - name: flip-test
+      taskSpec:
+        metadata:
+          annotations:
+            tekton.dev/template: ''
+          labels:
+            pipelines.kubeflow.org/cache_enabled: 'true'
+            pipelines.kubeflow.org/generation: ''
+            pipelines.kubeflow.org/pipelinename: ''
+        results:
+        - description: /tmp/output
+          name: output
+        steps:
+        - args:
+          - python -c "import random; result = 'heads' if random.randint(0,1) == 0
+            else 'tails'; print(result)" | tee $(results.output.path)
+          command:
+          - sh
+          - -c
+          image: python:alpine3.6
+          name: main
+      timeout: 0s
+    - name: flip-test-2
+      taskSpec:
+        metadata:
+          annotations:
+            tekton.dev/template: ''
+          labels:
+            pipelines.kubeflow.org/cache_enabled: 'true'
+            pipelines.kubeflow.org/generation: ''
+            pipelines.kubeflow.org/pipelinename: ''
+        results:
+        - description: /tmp/output
+          name: output
+        steps:
+        - args:
+          - python -c "import random; result = 'heads' if random.randint(0,1) == 0
+            else 'tails'; print(result)" | tee $(results.output.path)
+          command:
+          - sh
+          - -c
+          image: python:alpine3.6
+          name: main
+      timeout: 0s
+    - name: print-2
+      params:
+      - name: flip-test-output
+        value: $(tasks.flip-test.results.output)
+      runAfter:
+      - pipeline-flip-coin-graph-flip-component-1
+      taskSpec:
+        metadata:
+          annotations:
+            tekton.dev/template: ''
+          labels:
+            pipelines.kubeflow.org/cache_enabled: 'true'
+            pipelines.kubeflow.org/generation: ''
+            pipelines.kubeflow.org/pipelinename: ''
+        params:
+        - name: flip-test-output
+        steps:
+        - command:
+          - echo
+          - cool, it is over. $(inputs.params.flip-test-output)
+          image: alpine:3.6
+          name: main
+      timeout: 0s
+    - name: pipeline-flip-coin-graph-flip-component-1
+      params:
+      - name: flip-test-output
+        value: $(tasks.flip-test.results.output)
+      - name: just_one_iteration
+        value:
+        - '1'
+      - name: maxVal
+        value: $(params.maxVal)
+      runAfter:
+      - flip-test
+      - flip-test-2
+      taskRef:
+        apiVersion: custom.tekton.dev/v1alpha1
+        kind: PipelineLoop
+        name: pipeline-flip-coin-graph-flip-component-1
```

Comment:
This name doesn't match with the `recursion_while_pipelineloop_cr1.yaml` with the versioning tag. Right now all our loop resources has a versioning tag like `e023c ` to avoid resource conflicts.

Reply:
so looks like the code doesn't have any issue, but need to update the way how we generate the tests.

Reply:
Sure, will update and test.